networkx/linalg/bethehessianmatrix.py

```diff
@@ -68,9 +67,18 @@ def bethe_hessian_matrix(G, r=None, nodelist=None):
         nodelist = list(G)
     if r is None:
         r = sum(d ** 2 for v, d in nx.degree(G)) / sum(d for v, d in nx.degree(G)) - 1
-    A = nx.to_scipy_sparse_matrix(G, nodelist=nodelist, format="csr")
+    A = nx.to_scipy_sparse_array(G, nodelist=nodelist, format="csr")
     n, m = A.shape
-    diags = A.sum(axis=1)
-    D = sp.sparse.spdiags(diags.flatten(), [0], m, n, format="csr")
-    I = sp.sparse.eye(m, n, format="csr")
-    return (r ** 2 - 1) * I - r * A + D
+    # TODO: Rm csr_array wrapper when spdiags array creation becomes available
+    D = sp.sparse.csr_array(sp.sparse.spdiags(A.sum(axis=1), 0, m, n, format="csr"))
+    # TODO: Rm csr_array wrapper when eye array creation becomes available
```

Comment:
This is a nicely descriptive TODO.  Above there is one that simply says "TODO: csr_array"